### PR TITLE
Dispatcher changes2

### PR DESCRIPTION
**Pull request by TristanB22**
Hello! Here is a draft request for checking the types of things which are passed through the c union into the rustPosix dispatcher.

### CODE REVIEW

**Comment by TristanB22** [comment]
This is a draft PR for type checking. If you could please take a look at the get_cstarr function in types.rs, that is what I am most unsure about. Otherwise, I think that this is ready for merging if no one finds issues with it.

File: src/safeposix/dispatcher.rs [hunk 1]
```diff
@@ -181,22 +353,64 @@ pub extern "C" fn dispatcher(cageid: u64, callnum: i32, arg1: Arg, arg2: Arg, ar
             cage.getegid_syscall()
         }
         PREAD_SYSCALL => {
-            cage.pread_syscall(unsafe{arg1.dispatch_int}, unsafe{arg2.dispatch_mutcbuf}, unsafe{arg3.dispatch_usize}, unsafe{arg4.dispatch_isize})
+            match (interface::get_int(arg1), interface::get_mutcbuf(arg2), interface::get_usize(arg3), interface::get_isize(arg4)) {
+                (Ok(int1), Ok(mutcbuf2), Ok(usize3), Ok(isize4)) => {
+                    return cage.pread_syscall(int1, mutcbuf2, usize3, isize4);
+                }
+                (Err(returned_error_code), ..) |
```
Comment (rennergade) [inline src/safeposix/dispatcher.rs:360]: Let's order these first to last

**Review by rennergade — COMMENTED** [review]
Write a long form comment on the dispatcher to how the match statements/errors work. 

Looks great though!

**Comment by rennergade** [comment]
This looks great. Awesome job Tristan!

**Pull request merged**